cql3/restrictions/statement_restrictions.cc

```diff
@@ -969,5 +975,229 @@ bool single_column_restriction::LIKE::is_satisfied_by(bytes_view data, const que
     return (*_matcher)(data);
 }
 
+namespace wip {
+
+namespace {
+
+using cql3::selection::selection;
+
+/// Returns cdef's value from the fetched data.
+bytes_opt get_value(const column_definition* cdef, const selection& selection,
+                    const std::vector<bytes>& partition_key, const std::vector<bytes>& clustering_key,
+                    const std::vector<bytes_opt>& other_columns) {
+    switch (cdef->kind) {
+    case column_kind::partition_key:
+        return partition_key[cdef->id];
+    case column_kind::clustering_key:
+        return clustering_key[cdef->id];
+    case column_kind::static_column:
+    case column_kind::regular_column:
+        return other_columns[selection.index_of(*cdef)];
+    default:
+        throw exceptions::unsupported_operation_exception("Unknown column kind");
+    }
```

Comment:
Eventually, I'd like to see this also made part of the type system. So we'll have a partition_key_column expression object and the evaluation visitor visitor will visit it to fetch the value.